docs/contracts/errors.rst

```diff
@@ -0,0 +1,58 @@
+.. index:: ! error
+
+.. _errors:
+
+******
+Errors
+******
+
+Errors in Solidity provide a convenient and gas-efficient way to explain to the
+user why an operation failed. They can be defined inside and outside of contracts.
```

Comment:
I think it would be best to mention interfaces and libraries explicitly.

```suggestion
user why an operation failed. They can be defined inside and outside of contracts (including interfaces and libraries).
```

I'd also mention that global errors and errors from libraries can be used anywhere while errors from contracts and interfaces can only be used inside the contract and other contracts derived from it (is that correct?).

Reply:
No, that's not the case.